lib/ansible/plugins/strategy/__init__.py

```diff
@@ -666,8 +668,8 @@ def _process_pending_results(self, iterator, one_pass=False, max_passes=None):
                                 else:
                                     iterator.add_notification(original_host.name, notification)
                                     display.vv(f"Notification for handler {notification} has been saved.")
-                                break
-                            else:
+                                    break
```

Comment:
I think we actually do not need to `break` from this loop at all. The `search_handlers_by_notification` already provides the "last defined wins" functionality for handlers notified by name.

Reply:
I take that back. At this point we are interested only in whether any handler exists for the notification and then we do the actual notifying when `flush_handlers` is executed. But for the case when a handler does the notification, we need to know all handlers that matches and notify them right away.

So this does appear to be correct. Sorry for the noise :-)

Reply:
No worries at all! Thanks for the feedback :)